### PR TITLE
Allow viewers to see all docs links

### PR DESCRIPTION
**Pull request by jedcunningham**
Currently, Viewers can only see the "Documentation" link in the docs menu. This PR allows Viewers (and above) to see all of the links.

Before:
docs_links_before

After:
docs_links_after

### CODE REVIEW

File: airflow/security/permissions.py [hunk 1]
```diff
@@ -24,6 +24,10 @@
 RESOURCE_DAG_PREFIX = "DAG:"
 RESOURCE_DOCS_MENU = "Docs"
 RESOURCE_DOCS_LINK = "Documentation"
+RESOURCE_DOCS_WEBSITE = "Airflow Website"
+RESOURCE_DOCS_REPO = "GitHub Repo"
+RESOURCE_DOCS_API_SWAGGER = "REST API Reference (Swagger UI)"
+RESOURCE_DOCS_API_REDOC = "REST API Reference (Redoc)"
```
Comment (kaxil) [inline airflow/security/permissions.py]: Would be good to club them into a single `RESOURCE_DOCS_LINK` resource.

WDYT?

cc @jhtimmins

Comment (jedcunningham) [reply airflow/security/permissions.py]: I agree though I'm not sure if FAB will make that trivial. I'll experiment a bit this afternoon.

Comment (jedcunningham) [reply airflow/security/permissions.py]: @kaxil, take another look. Not sure if FABs behavior with a shared `name` is intentional or not which could be a risk.
This also leaves old resources around. Is that something we need to clean up?

File: airflow/www/security.py [hunk 2]
```diff
@@ -66,13 +66,13 @@ class AirflowSecurityManager(SecurityManager, LoggingMixin):  # pylint: disable=
...
         (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_DAG_RUN),
-        (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_DOCS_LINK),
-        (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_DOCS_MENU),
         (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_JOB),
         (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_AUDIT_LOG),
         (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_PLUGIN),
         (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_SLA_MISS),
         (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_TASK_INSTANCE),
+        (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_DOCS_MENU),
+        (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_DOCS_LINKS),
```
Comment (jhtimmins) [inline airflow/www/security.py:75]: These were alphabetized by resource name. Is there a reason we should move them?

File: airflow/www/extensions/init_appbuilder_links.py [hunk 3]
```diff
@@ -21,10 +21,19 @@
...
-    appbuilder.add_link("Documentation", href=get_docs_url(), category="Docs")
-    appbuilder.add_link("Airflow Website", href='https://airflow.apache.org', category="Docs")
-    appbuilder.add_link("GitHub Repo", href='https://github.com/apache/airflow', category="Docs")
+    appbuilder.add_link(name="docs_links", label="Documentation", href=get_docs_url(), category="Docs")
     appbuilder.add_link(
-        "REST API Reference (Swagger UI)", href='/api/v1./api/v1_swagger_ui_index', category="Docs"
+        name="docs_links", label="Airflow Website", href='https://airflow.apache.org', category="Docs"
+    )
+    appbuilder.add_link(
+        name="docs_links", label="GitHub Repo", href='https://github.com/apache/airflow', category="Docs"
```
Comment (kaxil) [inline airflow/www/extensions/init_appbuilder_links.py:29]: Yeah this does not feel right

Comment (kaxil) [reply airflow/www/extensions/init_appbuilder_links.py:29]: cc @jhtimmins any ideas for a better solution?

**Review by jhtimmins — COMMENTED** [review]
@kaxil @jedcunningham I actually don't mind the links sharing a name, nor do I mind them having different names. It's pretty clear that FAB internally assumes that the menu name will be used for the permission name, and we break away from that model within Airflow. So I don't think it's out of line to use the same name and just change the labels. Although if we go that route, I'd prefer we use `Documentation` or `Documentation Links` instead of `docs_links`, since this name will show up when different permissions are assigned to actual roles.

But since this likely isn't the explicit purpose of the FAB interface, I think it's also fine to use different permission names for the links. Although `Airflow Website` is a somewhat unclear name for a docs link. Those permission names should clearly reference docs.

If old permissions are left over, we'll need to add a database migration that removes the unused migrations.

File: airflow/www/security.py (hunk 2 above)
Comment (jedcunningham) [reply airflow/www/security.py:75]: I didn't realize it was intended to be alphabetical as audit log is out of order, but I now see the log -> audit log rename is the culprit here. I'll move these back and fix audit log too once we figure out how we want to proceed on the name issue.

**Comment by jedcunningham** [comment]
@kaxil @jhtimmins Either `Documentation Links` or unique names works for me. I think it boils down to whether there is a need for granularity. I could possibly see a desire to not want the REST API links shown if the admin hasn't exposed the REST API (though I still think showing what _could_ be available is a good default). Let me know how you'd like to proceed.

**Comment by kaxil** [comment]
> @kaxil @jhtimmins Either `Documentation Links` or unique names works for me. I think it boils down to whether there is a need for granularity. I could possibly see a desire to not want the REST API links shown if the admin hasn't exposed the REST API (though I still think showing what _could_ be available is a good default). Let me know how you'd like to proceed.

I don't think there is a need for more granularity tbh

File: airflow/www/extensions/init_appbuilder_links.py (hunk 3 above)
Comment (kaxil) [reply airflow/www/extensions/init_appbuilder_links.py:29]: Having a same name for links seems like a hack.

Can we not group them and have a single permission for it. Wdyt @jhtimmins

Comment (jhtimmins) [reply airflow/www/extensions/init_appbuilder_links.py:29]: @kaxil I mean, that's basically what this is. There isn't really another way to group these together.

**Comment by jhtimmins** [comment]
@kaxil @jedcunningham I actually think we should keep them separate bc of the issue with the API not being exposed, which Jed raised. Adding 4 permissions vs adding 1 seems like a pretty trivial difference, and it's gonna be confusing to users if they keep seeing links to API docs but have no ability to turn them off without losing all docs.

**Comment by kaxil** [comment]
I think having API docs is good in general, not all users will have all permissions for all endpoints anyways so why restrict the doc links

**Comment by jhtimmins** [comment]
@kaxil I'm mostly coming from a standpoint of personal preference. Beyond that I don't have a super strong opinion, so if that's what y'all prefer I'm down.

**Review by kaxil — APPROVED** [review]

**Pull request merged**